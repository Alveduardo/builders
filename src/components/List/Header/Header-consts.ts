export const HEADER = {
  TEST_ID: {
    LABEL: 'HEADER_LABEL_TEST_ID',
  },
};
