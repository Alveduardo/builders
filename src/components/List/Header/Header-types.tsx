import { TextStyle } from 'react-native';

interface DefaultStyles {
  labelStyle: TextStyle;
}

export interface HeaderStyles {
  defaultStyles: DefaultStyles;
}

type StylesConfig = DefaultStyles;

export interface HeaderConfig {
  styles: StylesConfig;
}

export interface HeaderProps {
  /**
   * The Header title.
   */
  title: string;
}
