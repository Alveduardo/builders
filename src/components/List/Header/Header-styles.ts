import { HeaderStyles } from './Header-types';

export const headerStyles: HeaderStyles = {
  defaultStyles: {
    labelStyle: {
      opacity: 0.8,
      marginBottom: 16,
      textAlign: 'center',
    },
  },
};
