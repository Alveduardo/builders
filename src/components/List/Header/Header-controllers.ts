import { headerStyles } from './Header-styles';

export const useHeaderConfig = () => {
  const {
    defaultStyles: { labelStyle },
  } = headerStyles;

  return {
    styles: {
      labelStyle,
    },
  };
};
