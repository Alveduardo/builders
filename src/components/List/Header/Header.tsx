import React from 'react';
import Label from '../../Label';
import { HEADER } from './Header-consts';
import { useHeaderConfig } from './Header-controllers';

const Header = ({ title = '' }) => {
  const {
    styles: { labelStyle },
  } = useHeaderConfig();

  return (
    <Label testID={HEADER.TEST_ID.LABEL} size='regular' style={labelStyle}>
      {title}
    </Label>
  );
};

export default Header;
