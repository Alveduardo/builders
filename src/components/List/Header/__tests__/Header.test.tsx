import React from 'react';
import { render } from '@testing-library/react-native';
import { HEADER } from '../Header-consts';
import Header from '../Header';

const headerPropsDefault = {
  title: 'title',
};

describe('Header ->', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(jest.fn());
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  describe('Snapshot ->', () => {
    it('Deve corresponder ao componente com as props default', () => {
      const tree = render(<Header {...headerPropsDefault} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Deve corresponder ao componente com as props mais utilizadas', () => {
      const tree = render(<Header {...headerPropsDefault} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('props', () => {
    describe('min', () => {
      it('Deve renderizar o componente exibindo a prop title', () => {
        const { getByTestId } = render(<Header {...headerPropsDefault} />);

        const component = getByTestId(HEADER.TEST_ID.LABEL);

        expect(component.children[0]).toBe(headerPropsDefault.title.toString());
      });
    });
  });
});
