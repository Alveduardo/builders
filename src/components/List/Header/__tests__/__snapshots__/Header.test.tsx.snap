// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header -> Snapshot -> Deve corresponder ao componente com as props default 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 18,
      },
      Object {
        "marginBottom": 16,
        "opacity": 0.8,
        "textAlign": "center",
      },
    ]
  }
  testID="HEADER_LABEL_TEST_ID"
>
  title
</Text>
`;

exports[`Header -> Snapshot -> Deve corresponder ao componente com as props mais utilizadas 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 18,
      },
      Object {
        "marginBottom": 16,
        "opacity": 0.8,
        "textAlign": "center",
      },
    ]
  }
  testID="HEADER_LABEL_TEST_ID"
>
  title
</Text>
`;
